gitgud/make_level.py

```diff
@@ -0,0 +1,114 @@
+import sys
+import os
+
+cwd = os.getcwd()
+
+def write_Init(skill_name, skill_path, level_name):
+    # If skills/skill/__init__.py doesn't exist, create a basic version (no levels)
+    if not os.path.exists(join(skill_path,"__init__.py")):
+        with open(os.path.join(skill_path,"__init__.py"), 'w+') as fp:
+            level_setup = "\n".join([
+                "import pkg_resources\n",
+                "from gitgud.skills.util import BasicLevel",
+                "from gitgud.skills.util import Skill\n"
+                "skill = Skill(",
+                "\t'{}',".format(skill_name),
+                "\t[",
+                "\t]",
+                ")",
+                ])
+            level_setup = level_setup.replace("\t", " " * 4)
+            fp.write(level_setup)
+        fp.close()
+        
+        # Read file
+        with open(os.path.join(os.path.join("gitgud","skills"),"__init__.py"), 'r') as fp:
+            filedata = fp.read()
+        fp.close()
+
+        # Add import statement
+        replace = "from gitgud.skills.{0} import skill as {0}_skill\n\nfrom gitgud.skills.u".format(skill_name)
+        filedata = filedata.replace("\nfrom gitgud.skills.u", replace)
+        
+        # Add to input array of Skill
+        replace = ",\n\t{}_skill\n]".format(skill_name)
+        filedata = filedata.replace("\n]", replace)
+        filedata = filedata.replace("\t", " " * 4)
+
+        # Write to file
+        with open(os.path.join(os.path.join("gitgud","skills"),"__init__.py"), 'w') as fp:
+            fp.write(filedata)
+        fp.close()
+        
+        write_Init(skill_name, skill_path, level_name)
+        print(fp.closed) #for debug, delete
+        return
+    else:
+        # Populate file with BasicLevel
+        with open(os.path.join(skill_path,"__init__.py"), 'r') as fp:
+            filedata = fp.read()
+        fp.close()
+
+        replace = ",\n\t\tBasicLevel('{0}', pkg_resources.resource_filename(__name__, '_{0}/'))\n\t]".format(level_name)
+        filedata = filedata.replace("\n\t]", replace)
+        filedata = filedata.replace("[,","[")
+        filedata = filedata.replace("\t", " " * 4)
+        
+        with open(os.path.join(skill_path,"__init__.py"), 'w') as fp:
+            fp.write(filedata)
+        fp.close()
+        print(fp.closed) #for debug, delete
+    return
+
+def main():
+    # Obtain input arguments
+    skill_name = sys.argv[1]
+    level_name = sys.argv[2]
+    
+    # Check if cwd is git-gud folder
+    if cwd[-7:] == "git-gud":
```

Comment:
I see your new change. Let me clarify, I actually think it's good for the file to be in the top level directory. The change I was referring to was on this line. Just check if there is `os.path.isdir(os.path.join(cwd, 'gitgud'))`

Reply:
Can you move it back and change this line?

Reply:
Changed

Reply:
some bugs hold off on review